pkg/vectorize/rpad/rpad.go

```diff
@@ -0,0 +1,73 @@
+// Copyright 2021 Matrix Origin
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package rpad
+
+import (
+	"github.com/matrixorigin/matrixone/pkg/container/types"
+)
+
+var (
+	RpadVarcharInt64 func(*types.Bytes, []int64, *types.Bytes) *types.Bytes
+	RpadCharInt64    func(*types.Bytes, []int64, *types.Bytes) *types.Bytes
+)
+
+func init() {
+	RpadVarcharInt64 = rpadInt64
+	RpadCharInt64 = rpadInt64
+}
+
+// rpadInt64 returns padded strings.
+// Specially, rpadInt64 returns "NULL" strings if lens[0] < 0
+func rpadInt64(strs *types.Bytes, lens []int64, padstrs *types.Bytes) *types.Bytes {
+	pad := padstrs.Get(0)
+	results := &types.Bytes{}
+
+	if lens[0] < 0 {
+		// return NULL if newLen<0
+		nullbytes := []byte("NULL")
```

Comment:
I think we use a bit map to store the NULL values, you may refer to pkg/container/nulls for some wrap up functions of this bitmap

Reply:
Sorry, I have fixed these problems. It may behaves like mysql now.
Here are what I have done:
- use `nulls.Nulls` to store NULL values
- support different combinations of parameter types and behave like mysql 8.0
- make rpad become multibyte-safe